### DISCUSSION TITLE
Describe all parameter fields $q_config

### DISCUSSION THREAD

**Discussion by homeandriy** [qtranslate/qtranslate-xt#1293] [General]
Is there a description of all the fields of the $q_config parameter somewhere in the plugin. And what are they responsible for? (Description of the fields and what they do in the plugin)

**Comment by herrvigg** [comment]
There is no simple answer to that question. It's always been a huge mess, coming from the legacy qTranslate.

`$q_config` is an interpretation of `$qtranslate_options` for which the default values are set here for front:
https://github.com/qtranslate/qtranslate-xt/blob/master/src/options.php#L100
and here for admin
https://github.com/qtranslate/qtranslate-xt/blob/master/src/admin/admin_options.php#L9

On top of that there are additional functions executed to set new key entries, which makes it even more complicated to track.
These massive dictionaries are a nightmare to maintain, also as they are stored with a huge bunch of option names.

Ideally there should be a unique class with clear fields. I've been doing refactoring to simplify some parts but there is still a lot to do. This should be improved before migrating to a new official plugin.
Third-party users should avoid interpreting these values directly because they are subject to change.

**Reply by homeandriy** [reply]
Can I help with plugin development? For example, describe the fields in the question above? or document any functions?

**Reply by herrvigg** [reply]
Thanks for proposing.
You can always submit a patch if you think it's useful. But this is more about refactoring the code to separate the logic between the options and the configuration, which contains many sub-parts. This should go to a separate class and all the globals should be removed, it will take some time to get there.
